code/modules/ntsl2/program.dm

```diff
@@ -0,0 +1,43 @@
+/datum/ntsl_program
+	var/id = -1
+	var/name = "NTSL Program"
+
+	var/obj/machinery/telecomms/server/S = null
+
+	New(var/my_id)
+		id = my_id
+
+		// Failsafe, kill any obsolete programs.
+		for(var/datum/ntsl_program/P in ntsl2.programs)
+			if(P.id == id)
+				P.kill()
+		..()
```

Comment:
```suggestion
		return ..()
```
I think that return values are important in `New()`?

Reply:
New() should never return a value.